### ISSUE TITLE
Processing of message 'al/createPackage' failed with error: 'Unexpected value 'ArrayAccess' of type 'Microsoft.Dynamics.Nav.CodeAnalysis.BoundKind''

### ISSUE THREAD

**Issue by MarcelNL** [Microsoft/AL#4428]
Hello,

when I compile 2000+ objects, I get the following error message:

```
[Error - 5:34:50 PM] Please report this issue to https://github.com/microsoft/al/issues including information on how to reproduce it, if possible.
Processing of message 'al/createPackage' failed with error: 'Unexpected value 'ArrayAccess' of type 'Microsoft.Dynamics.Nav.CodeAnalysis.BoundKind''
Details:
System.InvalidOperationException: Unexpected value 'ArrayAccess' of type 'Microsoft.Dynamics.Nav.CodeAnalysis.BoundKind'
   bij Microsoft.Dynamics.Nav.CodeAnalysis.Emit.CodeGenerator.EmitTextIndexAccessAssignmentExpression(BoundTextIndexAccess textIndex, ExpressionSyntax source) in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Emitter\CodeGen\CodeGenerator_Expressions.cs:regel 1575
   bij Microsoft.Dynamics.Nav.CodeAnalysis.Emit.CodeGenerator.EmitAssignmentExpression(TypeSymbol type, BoundExpression target, TypeSymbol sourceType, ExpressionSyntax sourceSyntax) in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Emitter\CodeGen\CodeGenerator_Expressions.cs:regel 1533
   bij Microsoft.Dynamics.Nav.CodeAnalysis.Emit.MethodCodeGenerator.EmitAssignmentStatement(BoundAssignmentStatement stmt) in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Emitter\CodeGen\MethodCodeGenerator_Statements.cs:regel 40
   bij Microsoft.Dynamics.Nav.CodeAnalysis.Emit.MethodCodeGenerator.EmitSingleStatement(ArrayBuilder`1 statementList, BoundStatement stmt) in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Emitter\CodeGen\MethodCodeGenerator_Statements.cs:regel 224
   bij Microsoft.Dynamics.Nav.CodeAnalysis.Emit.MethodCodeGenerator.EmitStatement(BoundStatement stmt) in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Emitter\CodeGen\MethodCodeGenerator_Statements.cs:regel 144
   bij Microsoft.Dynamics.Nav.CodeAnalysis.Emit.MethodCodeGenerator.EmitIfStatement(BoundIfStatement stmt) in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Emitter\CodeGen\MethodCodeGenerator_Statements.cs:regel 308
   bij Microsoft.Dynamics.Nav.CodeAnalysis.Emit.MethodCodeGenerator.EmitSingleStatement(ArrayBuilder`1 statementList, BoundStatement stmt) in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Emitter\CodeGen\MethodCodeGenerator_Statements.cs:regel 239
   bij Microsoft.Dynamics.Nav.CodeAnalysis.Emit.MethodCodeGenerator.EmitStatementList(BoundStatementList stmts) in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Emitter\CodeGen\MethodCodeGenerator_Statements.cs:regel 189
   bij Microsoft.Dynamics.Nav.CodeAnalysis.Emit.MethodCodeGenerator.EmitRepeatStatement(BoundRepeatStatement stmt) in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Emitter\CodeGen\MethodCodeGenerator_Statements.cs:regel 685
   bij Microsoft.Dynamics.Nav.CodeAnalysis.Emit.MethodCodeGenerator.EmitSingleStatement(ArrayBuilder`1 statementList, BoundStatement stmt) in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Emitter\CodeGen\MethodCodeGenerator_Statements.cs:regel 248
   bij Microsoft.Dynamics.Nav.CodeAnalysis.Emit.MethodCodeGenerator.EmitStatement(BoundStatement stmt) in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Emitter\CodeGen\MethodCodeGenerator_Statements.cs:regel 144
   bij Microsoft.Dynamics.Nav.CodeAnalysis.Emit.MethodCodeGenerator.EmitIfStatement(BoundIfStatement stmt) in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Emitter\CodeGen\MethodCodeGenerator_Statements.cs:regel 308
   bij Microsoft.Dynamics.Nav.CodeAnalysis.Emit.MethodCodeGenerator.EmitSingleStatement(ArrayBuilder`1 statementList, BoundStatement stmt) in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Emitter\CodeGen\MethodCodeGenerator_Statements.cs:regel 239
   bij Microsoft.Dynamics.Nav.CodeAnalysis.Emit.MethodCodeGenerator.EmitBlock(BoundBlock block) in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Emitter\CodeGen\MethodCodeGenerator_Statements.cs:regel 117
   bij Microsoft.Dynamics.Nav.CodeAnalysis.Emit.MethodCodeGenerator.EmitSingleStatement(ArrayBuilder`1 statementList, BoundStatement stmt) in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Emitter\CodeGen\MethodCodeGenerator_Statements.cs:regel 235
   bij Microsoft.Dynamics.Nav.CodeAnalysis.Emit.MethodCodeGenerator.EmitStatement(BoundStatement stmt) in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Emitter\CodeGen\MethodCodeGenerator_Statements.cs:regel 144
   bij Microsoft.Dynamics.Nav.CodeAnalysis.Emit.MethodCodeGenerator.EmitCaseLineStatement(BoundStatement stmt) in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Emitter\CodeGen\MethodCodeGenerator_Statements.cs:regel 799
   bij Microsoft.Dynamics.Nav.CodeAnalysis.Emit.MethodCodeGenerator.EmitCaseStatementAsIfThenElse(ArrayBuilder`1 statementList, BoundCaseStatement stmt) in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Emitter\CodeGen\MethodCodeGenerator_Statements.cs:regel 786
   bij Microsoft.Dynamics.Nav.CodeAnalysis.Emit.MethodCodeGenerator.EmitCaseStatement(ArrayBuilder`1 statementList, BoundCaseStatement stmt) in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Emitter\CodeGen\MethodCodeGenerator_Statements.cs:regel 740
   bij Microsoft.Dynamics.Nav.CodeAnalysis.Emit.MethodCodeGenerator.EmitSingleStatement(ArrayBuilder`1 statementList, BoundStatement stmt) in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Emitter\CodeGen\MethodCodeGenerator_Statements.cs:regel 254
   bij Microsoft.Dynamics.Nav.CodeAnalysis.Emit.MethodCodeGenerator.EmitBlock(BoundBlock block) in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Emitter\CodeGen\MethodCodeGenerator_Statements.cs:regel 117
   bij Microsoft.Dynamics.Nav.CodeAnalysis.Emit.MethodCodeGenerator.EmitSingleStatement(ArrayBuilder`1 statementList, BoundStatement stmt) in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Emitter\CodeGen\MethodCodeGenerator_Statements.cs:regel 235
   bij Microsoft.Dynamics.Nav.CodeAnalysis.Emit.MethodCodeGenerator.EmitStatement(BoundStatement stmt) in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Emitter\CodeGen\MethodCodeGenerator_Statements.cs:regel 144
   bij Microsoft.Dynamics.Nav.CodeAnalysis.Emit.MethodCodeGenerator.EmitForStatement(ArrayBuilder`1 statementList, BoundForStatement stmt) in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Emitter\CodeGen\MethodCodeGenerator_Statements.cs:regel 515
   bij Microsoft.Dynamics.Nav.CodeAnalysis.Emit.MethodCodeGenerator.EmitSingleStatement(ArrayBuilder`1 statementList, BoundStatement stmt) in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Emitter\CodeGen\MethodCodeGenerator_Statements.cs:regel 242
   bij Microsoft.Dynamics.Nav.CodeAnalysis.Emit.MethodCodeGenerator.EmitBlock(BoundBlock block) in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Emitter\CodeGen\MethodCodeGenerator_Statements.cs:regel 117
   bij Microsoft.Dynamics.Nav.CodeAnalysis.Emit.MethodCodeGenerator.EmitSingleStatement(ArrayBuilder`1 statementList, BoundStatement stmt) in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Emitter\CodeGen\MethodCodeGenerator_Statements.cs:regel 235
   bij Microsoft.Dynamics.Nav.CodeAnalysis.Emit.MethodCodeGenerator.EmitStatement(BoundStatement stmt) in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Emitter\CodeGen\MethodCodeGenerator_Statements.cs:regel 144
   bij Microsoft.Dynamics.Nav.CodeAnalysis.Emit.MethodCodeGenerator.EmitIfStatement(BoundIfStatement stmt) in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Emitter\CodeGen\MethodCodeGenerator_Statements.cs:regel 308
   bij Microsoft.Dynamics.Nav.CodeAnalysis.Emit.MethodCodeGenerator.EmitSingleStatement(ArrayBuilder`1 statementList, BoundStatement stmt) in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Emitter\CodeGen\MethodCodeGenerator_Statements.cs:regel 239
   bij Microsoft.Dynamics.Nav.CodeAnalysis.Emit.MethodCodeGenerator.EmitBlock(BoundBlock block) in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Emitter\CodeGen\MethodCodeGenerator_Statements.cs:regel 117
   bij Microsoft.Dynamics.Nav.CodeAnalysis.Emit.MethodCodeGenerator.EmitScopeRunMethod() in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Emitter\CodeGen\MethodCodeGenerator.cs:regel 656
   bij Microsoft.Dynamics.Nav.CodeAnalysis.Emit.MethodCodeGenerator.EmitScopeClass() in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Emitter\CodeGen\MethodCodeGenerator.cs:regel 275
   bij Microsoft.Dynamics.Nav.CodeAnalysis.Emit.MethodCodeGenerator.Generate() in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Emitter\CodeGen\MethodCodeGenerator.cs:regel 128
   bij Microsoft.Dynamics.Nav.CodeAnalysis.MethodCompiler.CompileMethod(MethodSymbol methodSymbol) in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Compiler\MethodCompiler.cs:regel 455
   bij Microsoft.Dynamics.Nav.CodeAnalysis.MethodCompiler.CompileMembers(ContainerSymbol containingType) in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Compiler\MethodCompiler.cs:regel 328
   bij Microsoft.Dynamics.Nav.CodeAnalysis.MethodCompiler.CompileObject(ObjectTypeSymbol containingType) in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Compiler\MethodCompiler.cs:regel 218
   bij Microsoft.Dynamics.Nav.CodeAnalysis.MethodCompiler.<>c__DisplayClass14_0.<CompileObjectAsTask>b__0() in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Compiler\MethodCompiler.cs:regel 202
   bij Microsoft.Dynamics.Nav.CodeAnalysis.Utilities.UICultureUtilities.<>c__DisplayClass5_0.<WithCurrentUICulture>b__0() in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Utilities\UICultureUtilities.cs:regel 152
   bij System.Threading.Tasks.Task.Execute()
--- Einde van stacktracering vanaf vorige locatie waar uitzondering is opgetreden ---
   bij System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   bij System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   bij Microsoft.Dynamics.Nav.CodeAnalysis.MethodCompiler.WaitForWorkers() in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Compiler\MethodCompiler.cs:regel 162
   bij Microsoft.Dynamics.Nav.CodeAnalysis.MethodCompiler.CompileMethodBodies(Compilation compilation, ModuleBuilder moduleBeingBuiltOpt, Boolean hasDeclarationErrors, DiagnosticBag diagnostics, Predicate`1 filterOpt, CancellationToken cancellationToken) in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Compiler\MethodCompiler.cs:regel 146
   bij Microsoft.Dynamics.Nav.CodeAnalysis.Compilation.CompileImpl(ModuleBuilder moduleBuilder, Boolean emittingPdb, DiagnosticBag diagnostics, Predicate`1 filterOpt, CancellationToken cancellationToken) in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Compilation\Compilation.cs:regel 886
   bij Microsoft.Dynamics.Nav.CodeAnalysis.Compilation.Compile(ModuleBuilder moduleBuilder, Boolean emittingPdb, DiagnosticBag diagnostics, Predicate`1 filterOpt, CancellationToken cancellationToken) in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Compilation\Compilation.cs:regel 861
   bij Microsoft.Dynamics.Nav.CodeAnalysis.CommandLine.CommonCompiler.PackageTheModule(Compilation compilation, CompilerTextWriter consoleOutput, CancellationToken cancellationToken, ErrorLogger errorLogger, AnalyzerDriver analyzerDriver) in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\CommandLine\CommonCompiler.cs:regel 577
   bij Microsoft.Dynamics.Nav.CodeAnalysis.CommandLine.CommonCompiler.RunCore(CompilerTextWriter consoleOutput, ErrorLogger errorLogger, CancellationToken cancellationToken) in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\CommandLine\CommonCompiler.cs:regel 435
   bij Microsoft.Dynamics.Nav.CodeAnalysis.CommandLine.CommonCompiler.Run(CompilerTextWriter consoleOutput, CancellationToken cancellationToken) in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\CommandLine\CommonCompiler.cs:regel 313
   bij Microsoft.Dynamics.Nav.EditorServices.Protocol.LanguageServer.Extensions.CreatePackageRequestHandler.<HandleAsync>d__4.MoveNext() in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.EditorServices.Protocol\LanguageServer\Extensions\CreatePackageRequestHandler.cs:regel 47
--- Einde van stacktracering vanaf vorige locatie waar uitzondering is opgetreden ---
   bij System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   bij System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   bij Microsoft.Dynamics.Nav.EditorServices.Protocol.MessageProtocol.RequestHandlerBase`1.<HandleAsync>d__10.MoveNext() in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.EditorServices.Protocol\MessageProtocol\RequestHandlerBase.cs:regel 45
--- Einde van stacktracering vanaf vorige locatie waar uitzondering is opgetreden ---
   bij System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   bij System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   bij Microsoft.Dynamics.Nav.EditorServices.Protocol.RequestRegistry.<Process>d__6.MoveNext() in C:\a\512\s\source\Prod\Microsoft.Dynamics.Nav.EditorServices.Protocol\Endpoints\RequestRegistry.cs:regel 59
```

**Versions:**
 - AL Langauge: 2.1.69331
 - Business Central: NL Dynamics NAV 13.1 (26678)

**Comment by atoader** [comment]
Hi @MarcelNL you should compile your extension with the ALC compiler with verbose logging enabled and check the log file for more information on what is causing this exception.

**Comment by atoader** [comment]
Hi Marcel! 
![image](https://user-images.githubusercontent.com/3097909/50899065-4f5ceb80-1411-11e9-8acc-e8585f2efd0f.png)
This is the correct option. 
First, make sure you are using the latest developer preview extension, not the one from the marketplace. 

Close VSCode, remove the EditorServicesLog.txt file from {UserFolder}\.vscode\extensions\microsoft.al-{Version}\bin, open your project.
When you encounter the error, stop VSCode and go back and open the EditorServicesLog.txt file and search for the exception.
Before the exception, you should see information about what the compiler was trying to achieve and hints about where in the code you can find the problem

**Comment by MarcelNL** [comment]
Hi Alexandru,

I have created a new december developer preview machine and downloaded the VSIX and followed the steps you mentioned. I noticed that the version from the preview is the same as the one from the marketplace:  2.1.69331.

The information I get from the EditorServiceLog does not give me more information to find the issue.

do I use the wrong version?

![editorservice](https://user-images.githubusercontent.com/46380504/50909263-c99a6980-142b-11e9-90fe-266bd24e9809.png)
